NDK_ROOT is undefined

I am using the DroneCodeSDK . But when I try to build for the android , I hit this command  `sudo make android instal` but I get this error
```Makefile:152: *** NDK_ROOT is undefined, please point the environment variable to android-ndk root..  Stop```.
 Although I have set the NDK_ROOT in .bashrc. Also i have set NDK_ROOT from command prompt . I am using Ubuntu. I am not able to build the code for android. Please suggest how to get rid of this error.

Yes , It is set to correct path and previous error are now fixed. But still I am getting some error while hitting the command "sudo make android install". This is the error " nstall configuration: "Debug"
-- Installing: /usr/local/lib/android/x86/libdronecode_sdk.so
CMake Error at core/cmake_install.cmake:36 (file):
  file INSTALL cannot copy file
  "/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/core/libdronecode_sdk.so"
  to "/usr/local/lib/android/x86/libdronecode_sdk.so".
Call Stack (most recent call first):
  cmake_install.cmake:38 (include)".
Please help me out to get the code compiled.

There are two things happening:- 
Case 1). When I run command as "sudo make android install". Then I get the NDK_ROOT error. 
 I have set the paths as below 
 export NDK_ROOT=/home/akash/Android/Sdk/ndk-bundle
 export ANDROID_TOOLCHAIN_CMAKE=/home/akash/Android/Sdk/ndk-bundle/build/cmake/android.toolchain.cmake
 export ANDROID_CMAKE_BIN=/home/akash/Android/Sdk/cmake/3.6.4111459/bin/cmake

I have ensured that the paths are set correctly by running echo$NDK_ROOT I have echo all three paths and yet they are pointing to correct location. But still I am getting the error "Makefile:152: *** NDK_ROOT is undefined, please point the environment variable to android-ndk root.. Stop."

Case 2). When I run the command without sudo as "make android install" then I get the following error
" file INSTALL cannot copy file
  "/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/core/libdronecode_sdk.so"
  to "/usr/local/lib/android/x86/libdronecode_sdk.so".
Call Stack (most recent call first):
  cmake_install.cmake:38 (include)
Makefile:93: recipe for target 'install' failed
make[1]: *** [install] Error 1
Makefile:86: recipe for target 'android_x86' failed
make: *** [android_x86] Error 2"

Please help me out to get this compiled.

I run the commands you mentioned, Below is the console output:- 

```CMake Error at plugins/mission/cmake_install.cmake:47 (file):
  file INSTALL cannot find
  "/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/plugins/mission/third_party/json11/libjson11.so".```.

The cmake code from line 47 is as below:- 

```if("${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
        file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/android/x86" TYPE FILE FILES "/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/plugins/mission/third_party/json11/libjson11.so")
        endif()

        if(NOT CMAKE_INSTALL_LOCAL_ONLY)
        # Include the install script for each subdirectory.
        include("/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/plugins/mission/third_party/json11/cmake_install.cmake")

        endif()```

Thanks.

Yes , I tried after a pull and now this is the newly introduced error from console output:- 

```Installing: /home/akash/Desktop/DroneCodeRetry/DronecodeSDK/install/lib/android/x86/libdronecode_sdk_mission.a
CMake Error at plugins/mission/cmake_install.cmake:47 (file):
  file INSTALL cannot find
  "/home/akash/Desktop/DroneCodeRetry/DronecodeSDK/build/android_x86/plugins/mission/third_party/json11/libjson11.so".
Call Stack (most recent call first):
  plugins/cmake_install.cmake:39 (include)
  cmake_install.cmake:39 (include)

--- a/core/log.cpp
+++ b/core/log.cpp
@@ -1,4 +1,5 @@
 #include "log.h"
+#include "global_include.h"
 
 #if defined(WINDOWS)
 #include "Windows.h"
@@ -43,6 +44,8 @@ void set_color(Color color)
             SetConsoleTextAttribute(handle, WIN_COLOR_RESET);
             break;
     }
+#elif defined(ANDROID)
+    UNUSED(color);
 #else
     switch (color) {
         case Color::RED:

--- a/core/log.h
+++ b/core/log.h
@@ -2,14 +2,14 @@
 
 #include <sstream>
 
-#if ANDROID
+#if defined(ANDROID)
 #include <android/log.h>
 #else
 #include <iostream>
 #include <ctime>
 #endif
 
-#ifndef WINDOWS
+#if !defined(WINDOWS)
 // Remove path and extract only filename.
 #define __FILENAME__ \
     (__builtin_strrchr(__FILE__, '/') ? __builtin_strrchr(__FILE__, '/') + 1 : __FILE__)

--- a/plugins/mission/CMakeLists.txt
+++ b/plugins/mission/CMakeLists.txt
@@ -34,7 +34,7 @@ install(TARGETS dronecode_sdk_mission
 )
 
 if(ANDROID)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/third_party/json11/libjson11.so
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/third_party/json11/libjson11.a
         DESTINATION ${lib_path}
     )
 endif()

